### PR TITLE
Boost demo now embeds vouchers in the GET request

### PR DESCRIPTION
**Pull request by lalexgap**
This updates the existing boost demo to support vouchers embedded in the request. 

Now when we make a request to `booster-http` we include the voucher in the request by including the voucher amount, channel id, and signature.

### CODE REVIEW

File: packages/boost-demo/src/App.tsx [hunk 1]
```diff
@@ -40,9 +42,11 @@ function App() {
   // If you're running this locally you'll need to override this value
```
Comment (lalexgap) [inline packages/boost-demo/src/App.tsx]: One nice thing about docker is that it includes a sample script that always generates the same  payload id. I should update the comment to acknowledge that.

File: packages/nitro-rpc-client/src/types.ts [hunk 2]
```diff
@@ -51,6 +51,14 @@ export type PaymentParams = {
   Amount: number;
   Channel: string;
 };
+
+export type Voucher = {
+  ChannelId: string;
+  // todo: this should be a bigint
```
Comment (geoknee) [inline packages/nitro-rpc-client/src/types.ts:57]: Shall we track this in a new issue?

File: packages/nitro-rpc-client/src/serde.ts [hunk 3]
```diff
@@ -98,14 +98,36 @@ const paymentSchema = {
...
 
+const voucherSchema = {
+  properties: {
+    ChannelId: { type: "string" },
+    Amount: { type: "uint32" },
+    Signature: {
+      type: "string",
+    },
+  },
+} as const;
```
Comment (geoknee) [inline packages/nitro-rpc-client/src/serde.ts:109]: Is it necessary to cast `as const`? This is already declared as a `const`...

File: packages/boost-demo/src/App.tsx [hunk 4]
```diff
@@ -82,49 +85,70 @@ function App() {
...
+
+      // This will prompt the browser to download the file
+      const blob = result.data;
+      const blobUrl = URL.createObjectURL(blob);
+      const a = document.createElement("a");
+      a.href = blobUrl;
+      a.download = "fetched-file-from-ipfs";
+      a.click();
+      a.remove();
+      window.URL.revokeObjectURL(blobUrl);
```
Comment (geoknee) [inline packages/boost-demo/src/App.tsx]: Could possibly be factored into a `triggerFileDownload` helper.

File: packages/boost-demo/src/App.tsx [hunk 5]
```diff
@@ -193,3 +232,16 @@ function App() {
...
+  let result = "0x";
+
+  const length = 132;
+  for (let i = 0; i < length; i++) {
+    const randomIndex = Math.floor(Math.random() * characters.length);
+    result += characters.charAt(randomIndex);
+  }
+
+  return result;
+}
```
Comment (geoknee) [inline packages/boost-demo/src/App.tsx]: Might be overkill for a "bad" signature? Could we just use all zeros, or hard code some random string?

**Review by geoknee — APPROVED** [review]

File: packages/nitro-rpc-client/src/serde.ts (hunk 3 above)
Comment (lalexgap) [reply packages/nitro-rpc-client/src/serde.ts:109]: I don't know why, but it seems necessary.  If I don't add the `as const`(like the other entries) I get a type error lower down. I'm sure if I spent some time relearning the typescript type system I could tell you why but I don't think I'm willing to do that 😛

File: packages/boost-demo/src/App.tsx (hunk 5 above)
Comment (lalexgap) [reply packages/boost-demo/src/App.tsx]: https://github.com/statechannels/nitro-gui/pull/88/commits/4d6121b1e7bf830c7e6b0bcf28cedf10200fd8a5

File: packages/boost-demo/src/App.tsx (hunk 4 above)
Comment (lalexgap) [reply packages/boost-demo/src/App.tsx]: https://github.com/statechannels/nitro-gui/pull/88/commits/68879f70ac9ec8957a411877ac1494e54ba443c1

**Pull request merged**